File: permutingTwoArrays/PermutingTwoArrays.java
package permutingTwoArrays;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class PermutingTwoArrays {
    public static void main(String[] args){
        List<Integer> A = new ArrayList<>();
        List<Integer> B = new ArrayList<>();
        Integer k = 10;
        A.addAll(Arrays.asList(3,2,4));
        B.addAll(Arrays.asList(6,7,8));
        permutingTwoArrays(k,A,B);
    }
    public static String permutingTwoArrays(Integer k ,List<Integer> A,List<Integer> B ){
        Collections.sort(A);
        Collections.sort(B  ,Comparator.reverseOrder());
        for(int i = 0; i < A.size(); i++){
            if(A.get(i) + B.get(i) < k ){return "NO";}
        }
        return "YES";
    }
}

File: permutingTwoArrays/permutingTwoArrays.ts
const permutingArrays2 = (k:number, a:number[], b:number[]):string => {
    a.sort((a, b) => a - b);
    b.sort((a, b) => b - a);
    for (let i = 0; i < a.length; i++) {
      if (a[i] + b[i] < k) return "NO";
    }
    return "YES";
  };
  
  const k2 = 10;
  const a2 = [4, 2, 4];
  const b2 = [6, 7, 8];
  
  console.log(permutingArrays2(k2, a2, b2));